clippy_lints/src/utils/conf.rs

```diff
@@ -110,8 +110,8 @@ macro_rules! define_Conf {
 define_Conf! {
     /// Lint: BLACKLISTED_NAME. The list of blacklisted names to lint about
     (blacklisted_names, "blacklisted_names", ["foo", "bar", "baz", "quux"] => Vec<String>),
-    /// Lint: CYCLOMATIC_COMPLEXITY. The maximum cyclomatic complexity a function can have
-    (cyclomatic_complexity_threshold, "cyclomatic_complexity_threshold", 25 => u64),
+    /// Lint: COGNITIVE_COMPLEXITY. The maximum cognitive complexity a function can have
+    (cognitive_complexity_threshold, "cognitive_complexity_threshold", 25 => u64),
```

Comment:
Can we somehow still accept the old name in config files or at least not hard error about it?

Reply:
I think that if I implement a formal name change like it was done in #3554, Clippy will tell you the name was changed.
More than that I don't know how to do, but that will be better than a hard error.

Reply:
I don't think this will work for the `clippy.toml` out of the box. But we don't have a mechanism in place for that yet.

There should be an unknown error for `cyclomatic_complexity`. Maybe this error can be extended to suggest the new name.
https://github.com/rust-lang/rust-clippy/blob/2141ebf69581ab3865ea8c83545e1941dd4d38af/tests/ui-toml/toml_unknown_key/conf_unknown_key.stderr#L1-L3

Or a new error type can be added here (this error is generated by the `utils::conf::read` function):
https://github.com/rust-lang/rust-clippy/blob/db13e6fa34707c62c5d87aa057813a40c67e29c4/clippy_lints/src/utils/conf.rs#L39-L44

Reply:
I think the key to change what error is generated lies in one of these two sections?

https://github.com/rust-lang/rust-clippy/blob/db13e6fa34707c62c5d87aa057813a40c67e29c4/clippy_lints/src/utils/conf.rs#L72-L79

https://github.com/rust-lang/rust-clippy/blob/db13e6fa34707c62c5d87aa057813a40c67e29c4/clippy_lints/src/utils/conf.rs#L110-L153

I'd like the error to match against `cyclomatic_complexity_threshold`, hopefully looking up in a list of renamed parameters, and if finding that, throwing a nicer "renaming error" rather than just the "generic name-not-found error"

Reply:
I have a slightly stupid idea/hack: readd the `(cyclomatic_complexity_threshold, "cyclomatic_complexity_threshold", 16544198613561 => u64),`

Note the overly specific `16544198613561`.

Then, have the code that reads the `cognitive_complexity_threshold` also read the `cyclomatic_complexity_threshold` and warn about an unused and outdated toml variabl if the value isn't `16544198613561`.